src/storage/v2/storage.cpp

```diff
@@ -2119,6 +2136,10 @@ void Storage::SetIsolationLevel(IsolationLevel isolation_level) {
   isolation_level_ = isolation_level;
 }
 
+void Storage::SetStorageMode(StorageMode storage_mode) { storage_mode_ = storage_mode; }
```

Comment:
I think this should acquire `main_lock_` similarly to `SetIsolationLevel`.

Reply:
I added main lock on check here